src/common/types/timestamp.cpp

```diff
@@ -306,37 +309,62 @@ timestamp_t Timestamp::FromEpochSeconds(int64_t sec) {
 
 timestamp_t Timestamp::FromEpochMs(int64_t ms) {
 	int64_t result;
+	if (!Timestamp::IsFinite(timestamp_t(ms))) {
+		return timestamp_t(ms);
+	}
 	if (!TryMultiplyOperator::Operation(ms, Interval::MICROS_PER_MSEC, result)) {
 		throw ConversionException("Could not convert Timestamp(MS) to Timestamp(US)");
 	}
 	return timestamp_t(result);
 }
 
 timestamp_t Timestamp::FromEpochMicroSeconds(int64_t micros) {
+	if (!Timestamp::IsFinite(timestamp_t(micros))) {
+		return timestamp_t(micros);
+	}
 	return timestamp_t(micros);
 }
 
 timestamp_t Timestamp::FromEpochNanoSeconds(int64_t ns) {
+	if (!Timestamp::IsFinite(timestamp_t(ns))) {
+		return timestamp_t(ns);
+	}
 	return timestamp_t(ns / 1000);
 }
 
 int64_t Timestamp::GetEpochSeconds(timestamp_t timestamp) {
+	if (!Timestamp::IsFinite(timestamp)) {
+		return timestamp.value;
+	}
 	return timestamp.value / Interval::MICROS_PER_SEC;
 }
 
 int64_t Timestamp::GetEpochMs(timestamp_t timestamp) {
+	if (!Timestamp::IsFinite(timestamp)) {
+		return timestamp.value;
+	}
 	return timestamp.value / Interval::MICROS_PER_MSEC;
 }
 
 int64_t Timestamp::GetEpochMicroSeconds(timestamp_t timestamp) {
+	if (!Timestamp::IsFinite(timestamp)) {
+		return timestamp.value;
+	}
 	return timestamp.value;
 }
 
-int64_t Timestamp::GetEpochNanoSeconds(timestamp_t timestamp) {
+int64_t Timestamp::GetEpochNanoSeconds(timestamp_t timestamp, const string &error_message) {
 	int64_t result;
 	int64_t ns_in_us = 1000;
+	if (!Timestamp::IsFinite(timestamp)) {
+		return timestamp.value;
+	}
 	if (!TryMultiplyOperator::Operation(timestamp.value, ns_in_us, result)) {
-		throw ConversionException("Could not convert Timestamp(US) to Timestamp(NS)");
+		if (error_message.empty()) {
```

Comment:
Perhaps the right thing to do here is to pass in a `const char *` for the units instead of a fully formatted error message?

Reply:
I was thinking of passing in an optional callback, that way the work only gets done when the error happens

Reply:
What we do in other places is to add a `TryGetEpochNanoSeconds` instead that returns a boolean and stores the result in a `timestamp_t &result`, then the caller can insert their own error message in case the function returns false.